src/Kinto.elm

```diff
@@ -64,22 +65,69 @@ type Endpoint
     | RecordEndpoint BucketName CollectionName RecordId
 
 
+type Filter
+    = Equal String String
+    | Min String String
+    | Max String String
+    | LT String String
+    | GT String String
+    | IN String (List String)
+    | NOT String String
+    | LIKE String String
+    | SINCE String
+    | BEFORE String
+
+
 type alias Body =
     Encode.Value
 
 
-type Request
-    = KintoRequest Config Endpoint Method
-    | KintoRequestWithBody Config Endpoint Method Body
+type alias Client =
+    { baseUrl : String
+    , headers : List ( String, String )
+    }
 
 
-type alias Config =
-    { baseUrl : String
-    , headers : List Http.Header
-    , auth : Auth
+type alias Resource a =
+    { itemEndpoint : String -> Endpoint
+    , listEndpoint : Endpoint
+    , itemDecoder : Decode.Decoder a
+    , listDecoder : Decode.Decoder (List a)
```

Comment:
I really like the way you've separated the verbs get/create/update/delete from the things being manipulated. Some random thoughts:

- Do we really need `listDecoder`? It seems like it should always just be equivalent to use `Decode.list resource.itemDecoder`.
- It's conceptually possible to have a `Resource (BucketEndpoint) (CollectionListEndpoint someBucket)`, although that doesn't make any sense. It's not actually possible for a user of this module to construct something like that, but it feels weird to me. I think fixing this would be a little annoying -- you'd have to separate singular endpoints from plural endpoints, and define URL mappings for both, so it might not be worth it.
- Going the other way, does the `Endpoint` type still have value? We no longer expose it, nor even need to expose it. I guess I still think it provides some conceptual clarity over just using strings..
- I feel like there's a little bit of ... redundancy, almost, between specifying a bucket, a collection, or a record. There's no way to link records to the collections that contain them, nor those collections to their buckets. But I can't think of any way to solve that which also provides any kind of value, so maybe that's fine.